drivers/watchdog/bflb_wdt.c

```diff
@@ -0,0 +1,297 @@
+// SPDX-License-Identifier: GPL-2.0
+
+#include <linux/err.h>
+#include <linux/init.h>
+#include <linux/module.h>
+#include <linux/kernel.h>
+#include <linux/utsname.h>
+#include <linux/io.h>
+#include <linux/of.h>
+#include <linux/platform_device.h>
+#include <linux/watchdog.h>
+
+#define BFLB_INT_TICKS_PER_SEC 1024
+#define BFLB_TICK_DIV 256
+#define BFLB_TICKS_PER_SEC ( BFLB_INT_TICKS_PER_SEC / BFLB_TICK_DIV )
+#define BFLB_MAX_SECS ( 65535 / BFLB_TICKS_PER_SEC )
+
+#define BFLB_DEFAULT_TIMEOUT 60
+
+#define BFLB_REG_BASE 0x2000A500
+
+#define BFLB_REG_WFAR 0x9C
+#define BFLB_VAL_WFAR 0xBABA
+
+#define BFLB_REG_WSAR 0xA0
+#define BFLB_VAL_WSAR 0xEB10
+
+#define BFLB_REG_WVR 0x6C
+#define BFLB_REG_WCR 0x98
+#define BFLB_VAL_WCR 0x1
```

Comment:
For WFAR and WSAR these really are magic numbers that come straight from the RM, but for the other registers with documented fields, it's better to define bit definitions that describe what the bits do, such as: "#define WCR_CNTR_RESET   BIT(0)"

Reply:
Fixed with [3a9cb51](https://github.com/arm000/linux-bl808/pull/8/commits/3a9cb5192e166b7c9d52b83390f723c6a6c2e071)